### PR TITLE
Implement API

### PR DESCRIPTION
**Pull request by kchodorek**
## Summary

Recently, we've been working on a Solidus-based project that uses Solidus via an API. To enable Stripe payments within this setup, we needed to add additional API endpoints and logic for creating payment intents within Stripe. The intent information can later be used in combination with the [Stripe Payment Element](https://docs.stripe.com/payments/payment-element).

This PR aims to integrate these changes into the upstream `solidus_stripe`.

It adds the following API endpoints:

- `Create Setup Intent`
- `Create Payment Intent`

More details on these endpoints can be found in the README [here](https://github.com/kchodorek/solidus_stripe/blob/a134d072ada24b3e606b7db309874a64241bc6c1/README.md#api-support).

The core logic of this change is encapsulated in service objects, making it easy to create these intents outside of the API as well (e.g., if you prefer a custom GraphQL mutation or need to create a payment intent outside of the regular checkout process for some reason).

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: app/models/solidus_stripe/customer.rb [hunk 1]
```diff
@@ -15,7 +15,21 @@ def self.retrieve_or_create_stripe_customer_id(payment_method:, order:)
...
 
     def create_stripe_customer
-      payment_method.gateway.request { Stripe::Customer.create(email: source.email) }
+      payment_method.gateway.request { Stripe::Customer.create(email: source.email, name: customer_name) }
+    end
+
+    def customer_name
+      return full_name if full_name.present?
+
+      source.email
```
Comment (jarednorman) [inline app/models/solidus_stripe/customer.rb]: This can be written simply as `full_name.presence || source.email` if you prefer.

**Review by jarednorman — COMMENTED** [review]
There's _a lot_ of mocking in these tests. I only skimmed this, but I didn't see anything resembling an integration test that tests if these new endpoints actually work as intended. 

I'll need to take a more serious look, but I'm concerned about the maintainability of this as is. I'd much prefer something where the controller tests aren't stubbing everything out and instead test that the objects work as arranged. I'd also try to pull the object construction up higher so there's less of the constructor stubbing code smell (`allow(SomeClass).to receive(:new)...`).

**Comment by kchodorek** [comment]
@jarednorman Thanks for your comment! I adjusted the specs, now instead of stubbing, I'm using `stripe-ruby-mock` gem, tests are much cleaner now.

File: bin/sandbox [hunk 2]
```diff
@@ -15,6 +15,7 @@ function unbundled {
 
 cd "./$app_name"
 unbundled bundle add $extension_name --path ..
+unbundled bundle add stripe-ruby-mock --group "test"
```
Comment (kennyadsl) [inline bin/sandbox]: Can you help me remember why we need this in the sandbox? I though having the dependency. in the dummy-app would be enough, but maybe I'm missing something.

**Review by kennyadsl — COMMENTED** [review]
Thanks for the contribution, left a comment but overall looks good! ❤️

File: bin/sandbox (hunk 2 above)
Comment (kchodorek) [reply bin/sandbox]: I wasn't sure if sandbox also suppose to include the new dependency - in such case I will remove it from `sandbox` and will leave it in `dummy-app`, thanks!

**Review by jarednorman — APPROVED** [review]
Thanks, that's much better.

**Review by kennyadsl — APPROVED** [review]

**Pull request merged**